components/json-splitter.js

```diff
@@ -0,0 +1,88 @@
+// json-splitter.js
+
+/**
+ * This component takes a json string with one or more array elements, parses it, and makes each array element into an 
+ * individual records.  Each record is adorned with a virtual node ID (vnid) to uniquely identify it in the rdf pipeline, 
+ * and sends them on to the next component.
+ */
+
+var jsonpointer = require('jsonpointer');
+
+var _ = require('underscore');
+var baseNode = require('./base-node');
+
+exports.getComponent = baseNode.updatePromise({
+
+  description: 'Splits a json dataset into individual records, adds a vnid to each, and sends them on.',
+
+  inPorts: { 
+      in: { 
+        addressable: true, 
+        datatype: 'string',
+        description: 'json string input',
+        required: true
+      },
+
+      json_pointer: {
+        addressable: true, 
+        datatype: 'string',
+        description: 'json pointer for the json data attribute to use as the vnid (RFC 6901)',
+        required: true
+      }
+  },
+
+  onchange: function( data ) {   
+
+      if ( this.in && this.json_pointer ) { 
+
+          var self = this;
+
+          var outPort = this.outPorts.out; 
+          var errorPort = this.outPorts.error; 
+
+          // Attempt to parse the JSON data received
+          var parsedData = null;
+          try { 
+             parsedData = JSON.parse( this.in.replace(/\r?\n|\r/,'') );
+          } catch(e) {
+            baseNode.handleError( "Unable to parse JSON for "+this.in+": "+e, 
+                                  errorPort );
+          }
+
+          var vnid; 
+          if ( _.isArray( parsedData ) ) { 
+
+              var jsonPointer = jsonpointer.compile(this.json_pointer); 
+              parsedData.forEach( function(datum) {   
+
+                  // Get and assign the vnid for this record
+                  vnid = jsonPointer.get(datum);
+                  if ( _.isNull( vnid ) ) 
+                      throw new Error( "Unable to find "+self.json_pointer+" in "+datum,
+                                       errorPort); 
+                  datum['vnid'] = vnid;
```

Comment:
I think the metadata (such as vnid) should be separated from the state (datum). Perhaps this should be changes to 
  outPort.send({vnid: vnid, state: datum})